### PR TITLE
Validation page 1

### PR DESCRIPTION
**Pull request by YanLabCHU**

### CODE REVIEW

**Review by jecos — APPROVED** [review]

File: src/main/java/bio/ferlab/clin/validation/utils/ValidatorUtils.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package bio.ferlab.clin.validation.utils;
+
+public class ValidatorUtils {
+    public static boolean isTrimmed(String value) {
+        if (value.length() == 0) {
+            return true;
+        }
+        return !Character.isWhitespace(value.charAt(0)) && !Character.isWhitespace(value.charAt(value.length() - 1));
```
Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/validation/utils/ValidatorUtils.java:8]: Do we need to validate if a string is trimmed?  Why not just call .trim() systematically?  And if we do, we simply have to trim the string prior to validating the length...

File: src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    public boolean validateResource(Patient patient) {
+        return validateNames(patient) &&
+                validateBirthDate(patient) &&
+                validateMRN(patient) &&
+                validateRAMQ(patient);
+    }
+
+    private boolean validateBirthDate(Patient patient) {
+        return patient.hasBirthDate() && patient.getBirthDate().before(new Date());
```
Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java:27]: .before( today ) or !after( today ) ?  Patient cannot be born on the same day?

File: src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    private boolean validateName(HumanName name) {
+        final String family = name.getFamilyElement().getValue();
+        final List<String> givens = name.getGiven().stream().map(StringType::getValue).collect(Collectors.toList());
+
+        return isValidName(family) && givens.stream().allMatch(this::isValidName);
+    }
+
+    private boolean isValidName(String name) {
+        return name != null &&
+                name.length() > 2 &&
```
Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java:74]: Greater than 2 or  >= 2?  I am pretty sure there are some asian patient with 2 letter names (ex.: Yu)

File: src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java (hunk 2 above)
Comment (plaplante-chust) [reply src/main/java/bio/ferlab/clin/validation/validators/PatientValidator.java:27]: If date sent includes hh:mm:ss, this works.  Then, please ignore.

**Pull request merged**